cpp/src/parquet/metadata.cc

```diff
@@ -1584,6 +1619,13 @@ class ColumnChunkMetaDataBuilder::ColumnChunkMetaDataBuilderImpl {
     return column_chunk_->meta_data.total_compressed_size;
   }
 
+  KeyValueMetadata& key_value_metadata() {
+    if (!key_value_metadata_) {
+      key_value_metadata_ = std::make_unique<KeyValueMetadata>();
+    }
+    return *key_value_metadata_;
```

Comment:
So once `key_value_metadata()` is called, the metadata is generated, and the metadata would be put in `SetKeyValueMetadata`?

Should we just add a "Put" here?